build.rs

```diff
@@ -1,56 +1,49 @@
+use std::{env, error::Error, fs, path::PathBuf};
+
+use git_version::git_version;
 use semver::Version;
-use std::{env, error::Error, fs, path::Path, path::PathBuf, process::Command};
 
 fn main() -> Result<(), Box<dyn Error>> {
-    // Put the linker script somewhere the linker can find it
-    let out = &PathBuf::from(env::var("OUT_DIR")?);
-    let mut linker_script = fs::read_to_string("defmt.x.in")?;
-    let hash = Command::new("git")
-        .args(&["rev-parse", "HEAD"])
-        .output()
-        .ok()
-        .and_then(|output| {
-            if output.status.success() {
-                String::from_utf8(output.stdout).ok()
-            } else {
-                None
-            }
-        });
-    let version = if let Some(hash) = hash {
-        hash
-    } else {
-        assert!(
-            !Path::new(".git").exists(),
-            "you need to install the `git` command line tool to use the git version of `defmt`"
-        );
-
-        // no git info -> assume crates.io
-        let semver = Version::parse(&std::env::var("CARGO_PKG_VERSION")?)?;
-        if semver.major == 0 {
+    const GIT_DESCRIBE: &str = git_version!(args = ["--long", "--match", "v*"], cargo_prefix = "");
```

Comment:
what happens here if the user doesn't have git installed (as previously caught by https://github.com/knurling-rs/defmt/pull/518/files#diff-d0d98998092552a1d3259338c2c71e118a5b8343dd4703c0c7f552ada7f9cb42L24)?

Reply:
Since we specify `cargo_prefix`, it will use the `CARGO_PKG_VERSION` (see the third bullet point https://docs.rs/git-version/0.3.4/git_version/macro.git_version.html)